crates/core/src/db/relational_db.rs

```diff
@@ -868,10 +944,124 @@ impl RelationalDB {
     }
 }
 
+#[allow(unused)]
+#[derive(Clone)]
+struct LockFile {
+    path: Arc<Path>,
+    lock: Arc<File>,
+}
+
+impl LockFile {
+    pub fn lock(root: impl AsRef<Path>) -> Result<Self, DBError> {
+        fs::create_dir_all(&root)?;
+        let path = root.as_ref().join("db.lock");
+        let lock = File::create(&path)?;
+        lock.try_lock_exclusive()
+            .map_err(|e| DatabaseError::DatabasedOpened(root.as_ref().to_path_buf(), e.into()))?;
+
+        Ok(Self {
+            path: path.into(),
+            lock: lock.into(),
+        })
+    }
+}
```

Comment:
We should merge this with `fs_utils::lockfile::Lockfile`. I'm fine to leave that as a follow-up, though.